script/install.sh

```diff
@@ -62,26 +62,54 @@ retry() {
         fi
     done
 }
-
-# url, path, [gpg signature url to verify]
+# path, verify (0/1), urls...
+# the gpg signature is assumed to be url+.sig
 download() {
-    local url path
-    url=$1
-    path=$2
-
-    retry curl "$url" -o "$path"
-    if [ ! -z "${3:-}" ]; then
-        verify "$3" "$path"
+    local path do_verify mirrors
+    path="$1"
+    do_verify="$2"
+    mirrors=("${@:3}")
+
+    try_all_mirrors() {
+        for i in "${!mirrors[@]}" ; do
+            if [ "$i" -gt 0 ] ; then
+                log "Falling back to mirror: ${mirrors[$i]}"
+            fi
+            if curl "${mirrors[$i]}" -o "$path" ; then
+                break
+            fi
+        done
+    }
+    retry try_all_mirrors
+    if [ "$do_verify" -eq 1 ]; then
+        verify "$path" "${mirrors[@]/%/.sig}"
     fi
 }
 
-# url
+# path, urls...
+download_with_verify() {
+    download "$1" 1 "${@:2}"
+}
+
+# path, urls...
+download_without_verify() {
+    download "$1" 0 "${@:2}"
+}
```

Comment:
I added these to functions to avoid the need of passing around a magic `1` or `0`